docker/Dockerfile

```diff
@@ -21,12 +28,31 @@ RUN mkdir -p experiment_runs/
 
 # Install Algorithmic efficiency repo
 RUN echo "Setting up algorithmic_efficiency repo"
-ARG GIT_URL=https://github.com/mlcommons/algorithmic-efficiency.git
-RUN git clone $GIT_URL && cd algorithmic-efficiency 
+ARG branch="main"
+ARG framework=""
+ARG git_url=https://github.com/mlcommons/algorithmic-efficiency.git
+RUN git clone $git_url && cd algorithmic-efficiency 
+RUN cd algorithmic-efficiency && git checkout $branch
 
 RUN cd algorithmic-efficiency && pip install -e '.[full]'
-RUN cd algorithmic-efficiency && pip install -e '.[jax_gpu]' -f 'https://storage.googleapis.com/jax-releases/jax_cuda_releases.html'
-RUN cd algorithmic-efficiency && pip install -e '.[pytorch_cpu]' -f 'https://download.pytorch.org/whl/torch_stable.html'
+
+RUN if [ "$framework" = "jax" ] ; then \
+        echo "Installing Jax GPU" \
+        && cd algorithmic-efficiency \
+        && pip install -e '.[jax_gpu]' -f 'https://storage.googleapis.com/jax-releases/jax_cuda_releases.html' \
+        && pip install -e '.[pytorch_cpu]' -f 'https://download.pytorch.org/whl/torch_stable.html'; \
+    elif [ "$framework" = "pytorch" ] ; then \
+        echo "Installing Pytorch GPU" \
+        && cd algorithmic-efficiency \
+        && pip install -e '.[jax_cpu]' \
+        && pip install -e '.[pytorch_gpu]' -f 'https://download.pytorch.org/whl/torch_stable.html'; \
+    else \
```

Comment:
maybe instead of this we should have "the default be "both", and this case is for "both" and the else case prints an error and exits

Reply:
good idea